src/app/components/HoldRequest/HoldRequest.jsx

```diff
@@ -26,31 +29,119 @@ class HoldRequest extends React.Component {
     this.setState({ data: Store.getState() });
   }
 
+  /**
+   * requireUser()
+   * Redirectes the patron to OAuth log in page if he/she hasn't been logged in yet.
+   *
+   * @return {Boolean}
+   */
   requireUser() {
-    if (!this.state.patron || !this.state.patron.id.length) {
-      const fullUrl = encodeURIComponent(window.location.href);
-      window.location.replace(`${config.loginUrl}?redirect_uri=${fullUrl}`);
-      return false;
+    if (this.state.patron && this.state.patron.id) {
+      return true;
     }
-    return true;
+
+    const fullUrl = encodeURIComponent(window.location.href);
+
+    window.location.replace(`${config.loginUrl}?redirect_uri=${fullUrl}`);
+
+    return false;
   }
 
-  render() {
-    const searchKeywords = this.props.searchKeywords;
-    const record = this.props.bib;
-    const title = record.title[0];
-    const bibId = record['@id'].substring(4);
-    const itemId = this.props.params.id;
-    const selectedItem = LibraryItem.getItem(record, itemId);
-    const location = LibraryItem.getLocation(record, itemId);
+  /**
+   * renderLoggedInInstruction(patronName)
+   * Renders the HTML elements and contents based on the patron data
+   *
+   * @param {String} patronName
+   * @return {HTML Element}
+   */
+  renderLoggedInInstruction(patronName) {
+    return (patronName) ?
+      <p className="loggedInInstruction">You are currently logged in as <strong>{patronName}</strong>. If this is not you, please <a href="https://isso.nypl.org/auth/logout">Log out</a> and sign in using your library card.</p>
+      : <p className="loggedInInstruction">Something wrong during retrieving your patron data.</p>;
+  }
 
+  render() {
+    const searchKeywords = this.state.data.searchKeywords || '';
+    const record = (this.state.data.item && !_isEmpty(this.state.data.item)) ?
+      this.state.data.item : null;
+    const title = (record && _isArray(record.title) && record.title.length) ?
+      record.title[0] : '';
+    const bibId = (record && record['@id'] && typeof record['@id'] === 'string') ?
+      record['@id'].substring(4) : '';
+    const patronName = (
+      this.state.patron.names && _isArray(this.state.patron.names) && this.state.patron.names.length
+      ) ? this.state.patron.names[0] : '';
+    const itemId = (this.props.params && this.props.params.id) ? this.props.params.id : '';
+    const selectedItem = (record && itemId) ? LibraryItem.getItem(record, itemId) : null;
+    const location = (record && itemId) ? LibraryItem.getLocation(record, itemId) : null;
     const monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July',
       'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];
-    let date = new Date();
+    const date = new Date();
     date.setDate(date.getDate() + 7);
     const day = date.getDate();
     const monthIndex = date.getMonth();
     const dateDisplay = `${monthNames[monthIndex]} ${day}`;
+    const content = (record) ? (
```

Comment:
Can you make this into an if/else statement? It'll be a bit easier to read that way.